packages/dashboards/addon/controllers/dashboards/dashboard/view.js

```diff
@@ -6,128 +6,130 @@
+      })
+      .serialize();
+
+    newFilter.dataSource = originalFilter.dimension.source;const index = newFilters.indexOf(filterToUpdate);
+    newFilters[index] = newFilter;
+
+    const filterQueryParams = await this.compression.compress({ filters: newFilters });
+
+    this.transitionToRoute('dashboards.dashboard', { queryParams: { filters: filterQueryParams } });
+  }
+
+  /**
+   * @action removeFilter
+   * @param {Object} dashboard
+   * @param {Object} filter
+   */
+  @action
+  async removeFilter(dashboard, filter) {
+    const filters = get(dashboard, 'filters').serialize();
+    const removedFilter = filter.serialize();
+    const newFilters = filters.filter(fil => !isEqual(fil, removedFilter));
+    const filterQueryParams = await get(this, 'compression').compress({ filters: newFilters });
+
+    this.transitionToRoute('dashboards.dashboard', { queryParams: { filters: filterQueryParams } });
+  }
+
+  /**
+   * @action addFilter
+   * @param {Object} dashboard
+   * @param {Object} dimension
+   */
+  @action
+  async addFilter(dashboard, dimension) {
+    const store = this.store;
+    const bardMetadata = this.metadataService;
+    const filters = dashboard.filters
+      .toArray()
+      .map(fil => {
         const newFil = fil.serialize();
         newFil.dataSource = fil.dimension.source;
         return newFil;
       }); //Native array of serialized filters
-      const dimensionMeta = bardMetadata.getById('dimension', dimension.dimension, dimension.dataSource);
-      const filter = store
-        .createFragment('bard-request/fragments/filter', {
-          dimension: dimensionMeta,
-          operator: 'in',
-          field: dimensionMeta.primaryKeyFieldName
-        })
-        .serialize();
+    const dimensionMeta = bardMetadata.getById('dimension', dimension.dimension, dimension.dataSource);
+    const filter = store
+      .createFragment('bard-request/fragments/filter', {
+        dimension: dimensionMeta,
+        operator: 'in',
+        field: dimensionMeta.primaryKeyFieldName
+      })
+      .serialize();
 
-      filter.dataSource = dimension.dataSource;
+    filter.dataSource = dimension.dataSource;
 
       filters.push(filter);
 
-      const filterQueryParams = await this.compression.compress({ filters });
+    const filterQueryParams = await this.compression.compress({ filters });
 
-      this.transitionToRoute('dashboards.dashboard', { queryParams: { filters: filterQueryParams } });
-    },
+    this.transitionToRoute('dashboards.dashboard', { queryParams: { filters: filterQueryParams } });
+  }
 
-    /**
-     * @action clearFilterQueryParams - Remove query params as model enters clean state
-     */
-    clearFilterQueryParams() {
-      this.set('filters', null);
-    }
+  /**
+   * @action clearFilterQueryParams - Remove query params as model enters clean state
+   */
+  @action
+  clearFilterQueryParams() {
+    this.set('filters', null);
```

Comment:
```suggestion
    this.filters = null;
```

Reply:
Query params are still managed through ember so we have to use set, I'll change it to be `set(...)`